lib/adapters/http.js

```diff
@@ -16,6 +16,33 @@ var enhanceError = require('../core/enhanceError');
 
 var isHttps = /https:?/;
 
+/**
+ *
+ * @param {http.ClientRequestArgs} options
+ * @param {AxiosProxyConfig} proxy
+ * @param {string} location
+ * @returns {http.ClientRequestArgs}
```

Comment:
If don't expect chain calls, I prefer to return nothing. Usually,

- `setSth(obj)`, modifies the argument directly
- `getSth(obj)`, returns a new cloned object